### PR TITLE
fix issues w run task args

### LINKED ISSUE #157814: Calling a task using TaskDefinition data does not work in 1.70.0
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.70.0
- OS Version: Darwin x64 21.6.0

Steps to Reproduce:

1. Provide a task to the `vscode.tasks.registerTaskProvider` wit the following data:
```js
    const vscodeTask = new vscode.Task(
      {
        type: "Foo",
        task: "foo.clean-project",
      },
       projectDir,
      "Clean project",
      "Foo",
      ....
    );
 ...
```
2. Run the task:

This code works:
```js
    vscode.commands.executeCommand('workbench.action.tasks.runTask', {
      type: "Foo",
      task: "Clean project",
    });
```

whereas, this does not:

```js
    vscode.commands.executeCommand('workbench.action.tasks.runTask', {
      type: "Foo",
      task: "foo.clean-project",
    });
```

### PR DESCRIPTION
**Pull request by meganrogge**
fixes #157814

### CODE REVIEW

File: src/vs/workbench/contrib/tasks/common/tasks.ts [hunk 1]
```diff
@@ -1226,7 +1226,6 @@ export namespace TaskDefinition {
 			// We have no task definition so we can't sanitize the literal. Take it as is
 			const copy = Objects.deepClone(external);
 			delete copy._key;
```
Comment (meganrogge) [inline src/vs/workbench/contrib/tasks/common/tasks.ts:1228]: return KeyedTaskIdentifier.create(copy);

Comment (meganrogge) [inline src/vs/workbench/contrib/tasks/common/tasks.ts:1228]:
```suggestion
			delete copy._key;
			return KeyedTaskIdentifier.create(copy);
```

**Review by chrmarti — DISMISSED** [review]

**Review by ProKashif — DISMISSED** [review]

File: src/vs/workbench/contrib/tasks/common/tasks.ts [hunk 2]
```diff
@@ -1226,7 +1226,6 @@ export namespace TaskDefinition {
...
-			return KeyedTaskIdentifier.create(copy);
 		}
```
Comment (meganrogge) [inline src/vs/workbench/contrib/tasks/common/tasks.ts:1229]:
```suggestion
return KeyedTaskIdentifier.create(copy);
		}
```

**Comment by Tyriar** [comment]
Some task quick pick smoke tests are still failing on Linux.

**Comment by meganrogge** [comment]
investigated and it looks like no tasks are getting found when run tasks is called or the quickpick is closing too soon before the second level quick pick is populated. tried delaying the close of the quickpick which didn't help, so creating a build to investigate and hopefully reproduce on WSL

**Comment by meganrogge** [comment]
investigated on WSL and saw that run task wasn't happening when there was no filter provided 😅

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 3]
```diff
@@ -2550,10 +2546,6 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 		}
 	}
 
-	private _canRunCommand(): boolean {
-		return true;
-	}
```
Comment (Tyriar) [inline src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:2555]: 🤔

File: test/automation/src/task.ts [hunk 4]
```diff
@@ -50,7 +50,7 @@ export class Task {
 				await this.terminal.assertSingleTab({ color: expected[0].icon.color, icon: expected[0].icon.id || 'tools' });
 			}
 		}
-		await this.quickinput.closeQuickInput();
+		await setTimeout(() => this.quickinput.closeQuickInput(), 500);
```
Comment (Tyriar) [inline test/automation/src/task.ts:53]: Timeouts like this are inherently flaky, is there some other condition we could wait on?

**Review by Tyriar — DISMISSED** [review]

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts (hunk 3 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:2555]: not sure what this did @alexr00

File: test/automation/src/task.ts (hunk 4 above)
Comment (meganrogge) [reply test/automation/src/task.ts:53]: I fixed the cause of the issue so this shouldn't be needed anymore at all

**Comment by Tyriar** [comment]
Conflict needs fixing

**Review by Tyriar — DISMISSED** [review]

**Review by Tyriar — APPROVED** [review]

**Pull request merged**